server/src/main/java/org/elasticsearch/action/admin/cluster/snapshots/status/SnapshotStatus.java

```diff
@@ -218,11 +235,113 @@ public XContentBuilder toXContent(XContentBuilder builder, Params params) throws
+                    XContentParserUtils.ensureExpectedToken(XContentParser.Token.VALUE_STRING, parser.nextToken(),
+                        parser::getTokenLocation);
+                    uuid = parser.text();
+                } else if (STATE.equals(currentFieldName)) {
+                    XContentParserUtils.ensureExpectedToken(XContentParser.Token.VALUE_STRING, parser.nextToken(),
+                        parser::getTokenLocation);
+                    String stateRaw = parser.text();
+                    try {
+                        state = SnapshotsInProgress.State.valueOf(stateRaw);
+                    } catch (IllegalArgumentException iae) {
+                        throw new ElasticsearchParseException("failed to parse snapshot status, unknown state value [{}]", iae, stateRaw);
+                    }
+                } else if (INCLUDE_GLOBAL_STATE.equals(currentFieldName)) {
+                    XContentParserUtils.ensureExpectedToken(XContentParser.Token.VALUE_BOOLEAN, parser.nextToken(),
+                        parser::getTokenLocation);
+                    includeGlobalState = parser.booleanValue();
+                } else if (SnapshotShardsStats.Fields.SHARDS_STATS.equals(currentFieldName)) {
+                    shardsStats = SnapshotShardsStats.fromXContent(parser);
+                } else if (SnapshotStats.Fields.STATS.equals(currentFieldName)) {
+                    stats = SnapshotStats.fromXContent(parser);
+                } else if (INDICES.equals(currentFieldName)) {
+                    XContentParserUtils.ensureExpectedToken(XContentParser.Token.START_OBJECT, parser.nextToken(),
+                        parser::getTokenLocation);
+                    while (parser.nextToken() != XContentParser.Token.END_OBJECT) {
+                        SnapshotIndexStatus indexStatus = SnapshotIndexStatus.fromXContent(parser);
+                        indices.put(indexStatus.getIndex(), indexStatus);
+                        shards.addAll(indexStatus.getShards().values());
+                    }
+                } else {
+                    throw new ElasticsearchParseException("failed to parse snapshot status, unknown field [{}]", currentFieldName);
+                }
+            } else {
+                throw new ElasticsearchParseException("failed to parse snapshot status");
+            }
+        }
+        if (name == null) {
+            throw new ElasticsearchParseException("failed to parse snapshot status, missing snapshot name");
+        } else if (uuid == null) {
+            throw new ElasticsearchParseException("failed to parse snapshot status, missing snapshot uuid");
+        } else if (repository == null) {
+            throw new ElasticsearchParseException("failed to parse snapshot status, missing snapshot repository");
+        } else if (state == null) {
+            throw new ElasticsearchParseException("failed to parse snapshot status, missing snapshot state");
+        }
```

Comment:
I don't know if we need this validation, given that we parse our own output and we know that info is going to be there